cirq-google/cirq_google/devices/serializable_device.py

```diff
@@ -108,6 +108,25 @@ def __init__(
         """
         self.qubits = qubits
         self.gate_definitions = gate_definitions
+        self._metadata = cirq.GridDeviceMetadata(
+            [
+                (pair[0], pair[1])
+                for gate_defs in gate_definitions.values()
+                for gate_def in gate_defs
+                if gate_def.number_of_qubits == 2
+                for pair in gate_def.target_set
+                if len(pair) == 2 and pair[0] < pair[1]
+            ],
+            cirq.Gateset(
+                *[g for g in gate_definitions.keys() if isinstance(g, (cirq.Gate, type(cirq.Gate)))]
```

Comment:
Why is this (cirq.Gate, type(cirq.Gate))?  This seems redundant but I assume there's a good reason for it?

Reply:
Full disclosure: Things in this module are in pretty rough shape. In tests, the type contracts are somehow getting violated in this constructor and other "non Type[_Gate]" things are making it into this function. I'm working on deprecating `qid_pairs` in `cirq-core` and there are a few hangups in `cirq_google` that this PR will help unblock.

High level is I want to make sure we keep `cirq-core` pristine and `cirq-google` devices limping along in whatever weird state it is in so that we can finish the items here #4744 and then make a more focused effort on cleaning this up knowing that `cirq-core` now gives a solid foundation.

This weirdness keeps `cirq-google`  behaving, albeit weirdly.